packages/file_selector/file_selector_android/android/src/main/java/io/flutter/plugins/file_selector/FileSelectorDelegate.java

```diff
@@ -101,13 +119,44 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
       case REQUEST_CODE_GET_DIRECTORY_PATH:
         handleGetDirectoryPathResult(resultCode, data);
         break;
+      case REQUEST_CODE_OPEN_FILE:
+        handleOpenFileResult(resultCode, data);
+        break;
       default:
         return false;
     }
 
     return true;
   }
 
+  @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+  private void launchGetDirectoryPath(@Nullable String initialDirectory) {
+    Intent getDirectoryPathIntent = new Intent(Intent.ACTION_OPEN_DOCUMENT_TREE);
+
+    if (initialDirectory != null && !initialDirectory.isEmpty()) {
+      Uri uri = getDirectoryPathIntent.getParcelableExtra("android.provider.extra.INITIAL_URI");
+      String scheme = uri.toString();
+      scheme = scheme.replace("/root/", initialDirectory);
+      uri = Uri.parse(scheme);
+      getDirectoryPathIntent.putExtra("android.provider.extra.INITIAL_URI", uri);
+    }
+    activity.startActivityForResult(getDirectoryPathIntent, REQUEST_CODE_GET_DIRECTORY_PATH);
+  }
+
+  private void launchOpenFile(boolean isMultipleSelection, String[] mimeTypes) {
+    Intent openFileIntent = new Intent(Intent.ACTION_GET_CONTENT);
+    openFileIntent.addCategory(Intent.CATEGORY_OPENABLE);
+    openFileIntent.setType(mimeTypes.length == 1 ? mimeTypes[0] : "*/*");
```

Comment:
i agree with the mimetypes, because right below we're checking for mimeTypes.length > 0, so we have to check also for <2

Reply:
The "*/*" is the mime type of files that are allowed, but since we also filter using "openFileIntent.putExtra(Intent.EXTRA_MIME_TYPES, mimeTypes);" with the specific mimeTypes there is no problem. I changed this to set "*/*" always.